app/shop.py

```diff
@@ -0,0 +1,40 @@
+import datetime
+from app.customer import Customer
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            products: dict,
+            location: list = None

```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]